pipeline-utils.groovy

```diff
@@ -169,4 +169,36 @@ def inside_assembler_container(args, fn) {
     }
 }
 
+// Send a notification when a job's status changes.
+// Treat non-SUCCESS states the same, so it does not send notifications when
+// changes go between UNSTABLE and FAILURE.
+def notify_status_change(build) {
+    def color = ''
+    def message = "<${env.BUILD_URL}|Build ${env.BUILD_NUMBER} of ${env.JOB_NAME}>"
+
+    if (build.currentResult == build.previousBuild?.result)
+        return
+
+    if (build.previousBuild?.result == null) {
+        echo 'The previous build is still running; ignoring its build state.'
+        return
+    } else if (build.currentResult == 'SUCCESS') {
+        message = ":partyparrot: ${message} is working again."
+        color = 'good'
+    } else if (build.previousBuild.result == 'SUCCESS') {
```

Comment:
I could filter previously aborted job states, but I'm not sure what to do about when the current job is aborted.  You apparently can't set the state to `ABORTED` yourself ([JENKINS-43339](https://issues.jenkins-ci.org/browse/JENKINS-43339)), so it could only be treated as either success or failure.  The sandbox gets angry about adding new fields to `currentBuild`, but maybe it could pass the `ABORTED` state through a different variable.